lib/libspl/include/Makefile.am

```diff
@@ -1,10 +1,9 @@
-SUBDIRS = ia32 rpc sys util
+SUBDIRS = ia32 rpc sys util os
 
 libspldir = $(includedir)/libspl
 libspl_HEADERS = \
 	$(top_srcdir)/lib/libspl/include/assert.h \
 	$(top_srcdir)/lib/libspl/include/atomic.h \
-	$(top_srcdir)/lib/libspl/include/devid.h \
```

Comment:
Can this file simple be removed entirely?  It was simply added on Linux to avoid changing some existing user space code.  With the in-progress refactoring any include from `os/linux/` code should be removed, and any new includes I'd expect to be part of platform specific code.

Reply:
If you don't mind me pruning dead `#if defined(__sun__) || defined(__sun)` code from libzpool and libzfs I will gladly nuke devid.h from existence :)

Reply:
I don't mind at all!  Let's go ahead and get rid of it.